Sources/Basics/Sandbox.swift

```diff
@@ -55,6 +53,29 @@ public enum Sandbox {
 // MARK: - macOS
 
 #if os(macOS)
+fileprivate let threadSafeDarwinCacheDirectories: ThreadSafeArrayStore<AbsolutePath> = {
```

Comment:
the use of ThreadSafeArrayStore here was for caching and we loose the cache with this change. that may be okay, but then we dont need ThreadSafeArrayStore

@abertelrud wdyt about loosing the cache?

Reply:
We do retain the caching in some aspect - the value will be computed once and re-used rather than computed each time.  The idea of replacing the `ThreadSafeArrayStore<AbsolutePath>` with `[AbsolutePath]` however still stands and would also simplify this further.  I think that need to wait for @abertelrud to chime in on that.

Reply:
I do think it's a good idea to compute this only once, but it does seem that the change looses the thread safety aspect.  Couldn't it be implemented the same way as before where the contents of ThreadSafeArrayStore are what get cached, rather than possibly having two separate ThreadSafeArrayStores get created racily?  i.e. keep the  property as static though no longer on an extension of TSCUtility.Platform.

Reply:
Actually, are the initializers of Swift statics atomic already?  If so, do we need `ThreadSafeArrayStore` at all here?

Reply:
I see — sorry, I read it as being inside the `public enum Sandbox {}` so a property of that enum.  In this case, it seems to me that the `ThreadSafeArrayStore` is unnecessary and we still get the compute-only-once behaviour.  @tomerd am I misunderstanding anything?  IIUC you said that this gets rid of the cache, but it doesn't sound as if it does.

Reply:
Based on this https://docs.swift.org/swift-book/LanguageGuide/Properties.html#ID264 it seems that the initialization is inherently atomic.

Reply:
yes I agree - ThreadSafeArrayStore is unnecessary here

Reply:
also please run with TSAN to make sure thread safety is okay (which is probably why we used ThreadSafeArrayStore in the first place

Reply:
A static property of the `Sandbox` type makes sense to me.

Reply:
I'm not sure that moving it into `Sandbox` makes it any more encapsulated, but does namespace it.  I don't see the value, but I can move it nonetheless.  I'll remove the now unnecessary use of `ThreadSafeArrayStore`.

Reply:
I have no strong opinion about whether it's a fileprivate global or a static inside Sandbox, so whatever @tomerd thinks is fine with me.

Reply:
Actually, that is even more odd - the value is used in a single free-standing function only.  I'm leaving it currently as a global until @tomerd responds.

Reply:
I prefer it inside Sandbox, but won't hold this back